scripts/bracket_orders.js

```diff
@@ -45,29 +48,82 @@ const argv = require("yargs")
+    console.log(">", error.response.data.error)
+  }
+  return price
+}
+
+const thresholdPercent = 2
+const isPriceReasonable = async function (exchange, targetTokenId, stableTokenId, price) {
+  const tokenInfo = await fetchTokenInfo(exchange, [targetTokenId, stableTokenId], artifacts)
+  const targetToken = tokenInfo[targetTokenId]
+  const stableToken = tokenInfo[stableTokenId]
+  const dexagPrice = await getDexagPrice(targetToken.symbol, stableToken.symbol)
+  if (dexagPrice === undefined) {
+    console.log("Warning: could not perform price check against dex.ag.")
+    const answer = await promptUser("Continue anyway? [yN] ")
+    if (answer != "y" && answer.toLowerCase() != "yes") {
+      return false
+    }
+  }
+  if (Math.abs(dexagPrice - price) >= thresholdPercent / 100) {
+    console.log("Warning: the chosen price differs by more than", thresholdPercent, "percent from the price found on dex.ag.")
+    console.log("         chosen price:", price, targetToken.symbol, "bought for 1", stableToken.symbol)
+    console.log("         dex.ag price:", dexagPrice, targetToken.symbol, "bought for 1", stableToken.symbol)
+    const answer = await promptUser("Continue anyway? [yN] ")
+    if (answer != "y" && answer.toLowerCase() != "yes") {
+      return false
+    }
+  }
+  return true
+}
+
 module.exports = async callback => {
   try {
+    await BatchExchange.setProvider(web3.currentProvider)
+    await BatchExchange.setNetwork(web3.network_id)
+    const exchange = await BatchExchange.deployed()
     const GnosisSafe = artifacts.require("GnosisSafe")
     const masterSafe = await GnosisSafe.at(argv.masterSafe)
 
-    console.log("Preparing order transaction data")
-    const transactionData = await buildOrderTransactionData(
-      argv.masterSafe,
-      argv.slaves,
-      argv.targetToken,
-      argv.stableToken,
-      argv.targetPrice,
-      web3,
-      artifacts,
-      true,
-      argv.priceRange,
-      argv.validFrom,
-      argv.expiry
-    )
+    // check price against dex.ag's API
+    const targetTokenId = argv.targetToken
+    const stableTokenId = argv.stableToken
+    const priceIsOk = await isPriceReasonable(exchange, targetTokenId, stableTokenId, argv.targetPrice)
+
+    if (priceIsOk) {
+      console.log("Preparing order transaction data")
+      const transactionData = await buildOrderTransactionData(
+        argv.masterSafe,
+        argv.slaves,
+        argv.targetToken,
+        argv.stableToken,
+        argv.targetPrice,
+        web3,
+        artifacts,
+        true,
+        argv.priceRange,
+        argv.validFrom,
+        argv.expiry
+      )
 
-    const answer = await promptUser("Are you sure you want to send this transaction to the EVM? [yN] ")
-    if (answer == "y" || answer.toLowerCase() == "yes") {
-      await signAndSend(masterSafe, transactionData, web3, argv.network)
+      const answer = await promptUser("Are you sure you want to send this transaction to the EVM? [yN] ")
```

Comment:
If the price is not okay, do we will send out the transaction? Probably this should go into an else branch

Reply:
`priceIsOk` is the final decision of the user, the price might as well be very unreasonable. I let the user send the transaction even if the price check fails because, for example, dex.ag might not have the traded token in their database but the token exists on the exchange.

Reply:
Ups, I was confused with the brackets... yeah makes sense